test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -7313,6 +7313,59 @@ BOOST_AUTO_TEST_CASE(no_address_members_on_contract)
 	CHECK_ERROR(text, TypeError, "Member \"delegatecall\" not found or not visible after argument-dependent lookup in contract");
 }
 
+
+BOOST_AUTO_TEST_CASE(valid_hex_with_underscores)
+{
+	char const* text = R"(
+			contract test {
+				uint256 a = 0x1234_abcd;
+				uint256 b = 0x1234_5678_9abc;
+				uint256 c = 0x123_4567;
+				uint256 d = 0x1234_56 wei;
```

Comment:
Hm, I wonder if we should actually disallow this in the next breaking release :)

Reply:
These tests need to be updated after #3628.

Reply:
@axic So will have to remove tests that includes an unit after the hex literal correct?